### PR TITLE
Etfe 2511 - commercial description page

### PR DESCRIPTION
**Pull request by funkmaster20th**
requires the goodstype in, and indexing, 
might refactor the message details into a helper,

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -914,3 +913,42 @@ itemBrandName.change.hidden = if goods are branded
...
 itemBrandName.error.brandName.length = Brand name must be {0} characters or less
 itemBrandName.error.brandName.invalid = Brand name must not include < and > and : and ;
+
+commercialDescription.title = Enter a commercial description for the {0}
+commercialDescription.heading = Enter a commercial description for the {0}
+commercialDescription.checkYourAnswersLabel = Commercial Description
+commercialDescription.error.required = Enter a commercial description of the goods
+commercialDescription.paragraph = This will be used to help identify the goods.
+commercialDescription.error.length = Commercial description must be 350 characters or less
+commercialDescription.change.hidden = changeCommercialDescription
```
Comment (ericohwotu) [inline conf/messages:923]: should this be 
```suggestion
commercialDescription.change.hidden = commercial description
```

File: conf/messages [hunk 2]
```diff
@@ -914,3 +913,42 @@ itemBrandName.change.hidden = if goods are branded
...
+commercialDescription.details.ethylAlcohol3 = if transporting multipacks state both the number of multipacks and number of individual units. For example: 5 x boxes of 6 x 35cl bottles of London dry gin
+commercialDescription.details.ethylAlcohol4 = - the spirit name (such as rum or vodka)
+commercialDescription.details.ethylAlcohol5 = - if the spirit drink is dry, blended or mixed
+commercialDescription.details.tobacco1 = Include any information that will help to identify the goods, such as:
+commercialDescription.details.tobacco2 = - the number of the lowest level of packaging used, such as individual cigarette packets or cigar boxes
+commercialDescription.details.tobacco3 = - if transporting multipacks state both the number of multipacks and number of individual units. For example: 200 x multipacks of 5 x 20 cigarettes
+commercialDescription.details.tobacco4 = - if transporting hand rolling or pipe tobacco state the number and weight of individual packets.
+commercialDescription.summary = Help with commercial description
+commercialDescription.error.xss = Document reference must not contain < and > and : and ;
```
Comment (ericohwotu) [inline conf/messages:953]: should this be Document reference?

File: test-utils/fixtures/messages/sections/items/CommercialDescriptionMessages.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+object CommercialDescriptionMessages {
+  sealed trait ViewMessages extends BaseMessages { _: i18n =>
+    def heading(goodsType: String) = s"Enter a commercial description for the $goodsType"
+    def title(goodsType: String): String = titleHelper(heading(goodsType))
+    val checkYourAnswersLabel = "Commercial Description"
+    val errorRequired = "Enter a commercial description of the goods"
+    val paragraph = "This will be used to help identify the goods."
+    val errorlength = "Commercial description must be 350 characters or less"
+    val  changehidden = "changeCommercialDescription"
```
Comment (ericohwotu) [inline test-utils/fixtures/messages/sections/items/CommercialDescriptionMessages.scala:29]:
```suggestion
    val  changehidden = "commercial description"
```

File: test/controllers/sections/items/CommercialDescriptionControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+ * limitations under the License.
+ */
+
+package controllers.sections.items
+
+import base.SpecBase
+import forms.sections.items.CommercialDescriptionFormProvider
+import mocks.services.MockUserAnswersService
+import models.GoodsTypeModel.Wine
+import models.{Index, NormalMode, UserAnswers}
```
Comment (ericohwotu) [inline test/controllers/sections/items/CommercialDescriptionControllerSpec.scala:23]: unused import UserAnswers

File: test/controllers/sections/items/CommercialDescriptionControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+import navigation.FakeNavigators.FakeItemsNavigator
+import navigation.{ItemsNavigator, Navigator}
```
Comment (ericohwotu) [inline test/controllers/sections/items/CommercialDescriptionControllerSpec.scala:25]: Unused import Navigator

File: test/forms/sections/items/CommercialDescriptionFormProviderSpec.scala [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  val lengthKey = "commercialDescription.error.length"
+  val maxLength = 350
+
+  val form = new CommercialDescriptionFormProvider()()
+
+  ".value" - {
+
+    val fieldName = "value"
+
+    behave like fieldThatBindsValidData(
```
Comment (ericohwotu) [inline test/forms/sections/items/CommercialDescriptionFormProviderSpec.scala:34]: Do we need one for the XSS

File: test/navigation/ItemsNavigatorSpec.scala [hunk 7]
```diff
@@ -35,9 +37,19 @@ class ItemsNavigatorSpec extends SpecBase {
 
       "must go from the Item Brand Name page" - {
 
+        "to the Commercial Description Page" in {
+          val userAnswers = emptyUserAnswers.set(ItemBrandNamePage(testIndex1), ItemBrandNameModel(hasBrandName = true, Some("brand")))
+
+
+          navigator.nextPage(ItemBrandNamePage(testIndex1), NormalMode, userAnswers) mustBe
+            itemsRoutes.CommercialDescriptionController.onPageLoad(testErn, testDraftId, Index(0), NormalMode)
```
Comment (ericohwotu) [inline test/navigation/ItemsNavigatorSpec.scala:45]: [Optional] `Index(0)` can be changed to `testIndex1`

File: test/viewmodels/checkAnswers/sections/items/CommercialDescriptionSummarySpec.scala [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+package viewmodels.checkAnswers.sections.items
+
+import base.SpecBase
+import fixtures.messages.sections.items.CommercialDescriptionMessages
+import models.CheckMode
+import org.scalatest.matchers.must.Matchers
+import pages.sections.items.CommercialDescriptionPage
+import play.api.i18n.Messages
+import play.api.test.FakeRequest
+import uk.gov.hmrc.govukfrontend.views.Aliases.{HtmlContent, Text, Value}
```
Comment (ericohwotu) [inline test/viewmodels/checkAnswers/sections/items/CommercialDescriptionSummarySpec.scala:26]: unused import HTMLContent

File: test/views/sections/items/CommercialDescriptionViewSpec.scala [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+ * limitations under the License.
+ */
+
+package views.sections.items
+
+import base.ViewSpecBase
+import fixtures.messages.sections.items.CommercialDescriptionMessages
+import forms.sections.items.CommercialDescriptionFormProvider
+import models.GoodsTypeModel.Beer
+import models.NormalMode
```
Comment (ericohwotu) [inline test/views/sections/items/CommercialDescriptionViewSpec.scala:23]: Done believe this is being used

File: test/views/sections/items/CommercialDescriptionViewSpec.scala [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+import views.html.sections.items.CommercialDescriptionView
+import views.{BaseSelectors, ViewBehaviours}
+
+class CommercialDescriptionViewSpec extends ViewSpecBase with ViewBehaviours {
+
+  object Selectors extends BaseSelectors
+
+  "CommercialDescriptionView" - {
+
+    Seq(CommercialDescriptionMessages.English).foreach { messagesForLanguage =>
```
Comment (ericohwotu) [inline test/views/sections/items/CommercialDescriptionViewSpec.scala:39]: Would it be worth adding tests for more than one of the good types.

Comment (ericohwotu) [reply test/views/sections/items/CommercialDescriptionViewSpec.scala:39]: Nevermind the Select type page is not in yet

Comment (funkmaster20th) [reply test/views/sections/items/CommercialDescriptionViewSpec.scala:38]: added enegry and wine

File: test/views/sections/items/CommercialDescriptionViewSpec.scala (hunk 9 above)
Comment (funkmaster20th) [reply test/views/sections/items/CommercialDescriptionViewSpec.scala]: removed

File: test/viewmodels/checkAnswers/sections/items/CommercialDescriptionSummarySpec.scala (hunk 8 above)
Comment (funkmaster20th) [reply test/viewmodels/checkAnswers/sections/items/CommercialDescriptionSummarySpec.scala:26]: removed

File: test/navigation/ItemsNavigatorSpec.scala (hunk 7 above)
Comment (funkmaster20th) [reply test/navigation/ItemsNavigatorSpec.scala]: changed

File: test/forms/sections/items/CommercialDescriptionFormProviderSpec.scala (hunk 6 above)
Comment (funkmaster20th) [reply test/forms/sections/items/CommercialDescriptionFormProviderSpec.scala:38]: added section for errors

File: test/controllers/sections/items/CommercialDescriptionControllerSpec.scala (hunk 5 above)
Comment (funkmaster20th) [reply test/controllers/sections/items/CommercialDescriptionControllerSpec.scala]: removed

File: test-utils/fixtures/messages/sections/items/CommercialDescriptionMessages.scala (hunk 3 above)
Comment (funkmaster20th) [reply test-utils/fixtures/messages/sections/items/CommercialDescriptionMessages.scala]: changed

File: test/controllers/sections/items/CommercialDescriptionControllerSpec.scala (hunk 4 above)
Comment (funkmaster20th) [reply test/controllers/sections/items/CommercialDescriptionControllerSpec.scala]: removed

File: conf/messages (hunk 2 above)
Comment (funkmaster20th) [reply conf/messages]: it should not be changed

File: conf/messages (hunk 1 above)
Comment (funkmaster20th) [reply conf/messages]: changed

**Review by ericohwotu — APPROVED** [review]

**Review by ericohwotu — APPROVED** [review]

**Pull request merged**